src/SmartFormat/Extensions/ReflectionSource.cs

```diff
@@ -16,23 +16,53 @@ namespace SmartFormat.Extensions;
 /// Class to evaluate any <see cref="object"/> using <see cref="System.Reflection"/>.
 /// A type cache is used in order to reduce reflection calls.
 /// Include this source, if any of these types shall be used.
+/// <para/>
+/// For <see cref="SmartSettings.CaseSensitivity"/>==<see cref="CaseSensitivityType.CaseInsensitive"/>,
+/// when there are multiple members with the same name but different case, the first member is used.
 /// </summary>
 public class ReflectionSource : Source
 {
     private static readonly object[] Empty = Array.Empty<object>();
 
+    private static int _maxCacheSize = DefaultCacheSize;
+
+    /// <summary>
+    /// The default cache size.
+    /// </summary>
+    public const int DefaultCacheSize = 500;
```

Comment:
After unit tests the cache is filled with about 200 items - so 500 gives a good safety margin?